ndonnx/_array.py

```diff
@@ -120,7 +120,7 @@ def dynamic_size(self) -> Array:
         if self.ndim == 0:
             Array._from_tyarray(onnx.const(1, dtype=onnx.int64))
         if self.ndim == 1:
-            return self.dynamic_shape
+            return self.dynamic_shape[0]
 
         return Array._from_tyarray(self.dynamic_shape._tyarray.prod())
```

Comment:
We have a scalar type right?

Reply:
I'm not sure I understand what you mean. Indexing into an array-api-array always returns an array, but that array may be a rank-0 one (as is the case here). We don't have an explicit type for that though.

Reply:
That's right, I briefly mistook the utility of PyScalar.